src/hotspot/share/gc/z/zInitialize.hpp

```diff
@@ -24,16 +24,38 @@
 #ifndef SHARE_GC_Z_ZINITIALIZE_HPP
 #define SHARE_GC_Z_ZINITIALIZE_HPP
 
-#include "memory/allocation.hpp"
+#include "memory/allStatic.hpp"
+#include "utilities/compilerWarnings.hpp"
+
+#include <cstddef>
 
 class ZBarrierSet;
 
-class ZInitialize {
+class ZInitializer {
+ public:
+  ZInitializer(ZBarrierSet* barrier_set);
+};
+
+class ZInitialize : public AllStatic {
 private:
-  void pd_initialize();
+  static constexpr size_t ErrorMessageLength = 256;
+  static char _error_message[ErrorMessageLength];
```

Comment:
I would suggest that we separate the constants
```suggestion
  static constexpr size_t ErrorMessageLength = 256;

  static char _error_message[ErrorMessageLength];
```